GeneratorInterface/Pythia8Interface/plugins/JetMatchingEWKFxFx.cc

```diff
@@ -0,0 +1,997 @@
+            "JetMatchingMadgraph:init: No ickkw");
+      if (!par.haveParam("maxjetflavor"))
+        errorMsg(
+            "Warning in "
+            "JetMatchingMadgraph:init: No maxjetflavor");
+      if (!par.haveParam("alpsfact"))
+        errorMsg(
+            "Warning in "
+            "JetMatchingMadgraph:init: No alpsfact");
+    }
+  }
+
+  // Read in FxFx matching parameters
+  doFxFx = settingsPtr->flag("JetMatching:doFxFx");
+  nPartonsNow = settingsPtr->mode("JetMatching:nPartonsNow");
+  qCutME = settingsPtr->parm("JetMatching:qCutME");
+  qCutMESq = pow(qCutME, 2);
+
+  // Read in Madgraph merging parameters
+  doMerge = settingsPtr->flag("JetMatching:merge");
+  doShowerKt = settingsPtr->flag("JetMatching:doShowerKt");
+  qCut = settingsPtr->parm("JetMatching:qCut");
+  nQmatch = settingsPtr->mode("JetMatching:nQmatch");
+  clFact = settingsPtr->parm("JetMatching:clFact");
+
+  // Read in jet algorithm parameters
+  jetAlgorithm = settingsPtr->mode("JetMatching:jetAlgorithm");
+  nJetMax = settingsPtr->mode("JetMatching:nJetMax");
+  eTjetMin = settingsPtr->parm("JetMatching:eTjetMin");
+  coneRadius = settingsPtr->parm("JetMatching:coneRadius");
+  etaJetMax = settingsPtr->parm("JetMatching:etaJetMax");
+  slowJetPower = settingsPtr->mode("JetMatching:slowJetPower");
+
+  // Matching procedure
+  jetAllow = settingsPtr->mode("JetMatching:jetAllow");
+  exclusiveMode = settingsPtr->mode("JetMatching:exclusive");
+  qCutSq = pow(qCut, 2);
+  etaJetMaxAlgo = etaJetMax;
+  //performVeto    = true;
+
+  // If not merging is applied, then we are done
+  if (!doMerge)
+    return true;
+
+  // Exclusive mode; if set to 2, then set based on nJet/nJetMax
+  if (exclusiveMode == 2) {
+    // No nJet or nJetMax, so default to exclusive mode
+    if (nJetMax < 0) {
+      errorMsg(
+          "Warning in JetMatchingMadgraph:init: "
+          "missing jet multiplicity information; running in exclusive mode");
+      exclusiveMode = 1;
+    }
+  }
+
+  // Initialise chosen jet algorithm.
+  // Currently, this only supports the kT-algorithm in SlowJet.
+  // Use the QCD distance measure by default.
+  jetAlgorithm = 2;
+
+  slowJet = new SlowJet(slowJetPower, coneRadius, eTjetMin, etaJetMaxAlgo, 2, 2, nullptr, false);
+
+  // For FxFx, also initialise jet algorithm to define matrix element jets.
+  slowJetHard = new SlowJet(slowJetPower, coneRadius, qCutME, etaJetMaxAlgo, 2, 2, nullptr, false);
+
+  // To access the DJR's
+  slowJetDJR = new SlowJet(slowJetPower, coneRadius, qCutME, etaJetMaxAlgo, 2, 2, nullptr, false);
+
+  // A special version of SlowJet to handle heavy and other partons
+  hjSlowJet = new HJSlowJet(slowJetPower, coneRadius, 0.0, 100.0, 1, 2, nullptr, false, true);
+
+  // Setup local event records
+  eventProcessOrig.init("(eventProcessOrig)", particleDataPtr);
+  eventProcess.init("(eventProcess)", particleDataPtr);
+  workEventJet.init("(workEventJet)", particleDataPtr);
+
+  // Print information
```

Comment:
`cout` is not thread sale: please think about protecting it with the `MATCHINGDEBUG` directive, as you do in all other cases, or move to some MessageLogger report instead